compiler/noirc_frontend/src/tests/visibility.rs

```diff
@@ -146,3 +146,146 @@ fn does_not_error_if_calling_private_struct_function_from_same_module() {
     "#;
     assert_no_errors(src);
 }
+
+#[test]
+fn error_when_accessing_private_struct_field() {
```

Comment:
Are we covering the cases where the field is more public than the struct itself, i.e.
```noir
struct Foo {
    pub x: Field,
}
```
or
```noir
pub(crate) struct Foo {
    pub x: Field,
}
```

Reply:
Ah, good point. I didn't include those checks in this PR because we are missing those in other places. For example for this code:

```rust
struct Foo {}

pub fn bar() -> Foo {
    Foo {}
}
```

Rust says that `Foo` is more private than `bar`.

Would it be okay to capture an issue for all of these and fix them in a separate PR?

Reply:
> Would it be okay to capture an issue for all of these and fix them in a separate PR?

Yes, definitely 👍

Reply:
Done! https://github.com/noir-lang/noir/issues/6248